nameko/testing/pytest.py

```diff
@@ -182,8 +183,9 @@ def rabbit_uri(request, vhost_pipeline):
 
 @pytest.yield_fixture
 def rabbit_config(rabbit_uri):
-    with config.patch({'AMQP_URI': rabbit_uri}):
-        yield
+    conf = {'AMQP_URI': rabbit_uri}
+    with config.patch(conf):
+        yield conf  # nameko 2.X backward compatible
```

Comment:
yielding just config may not work on places where the dictionary is changed in tests using the fixture I think. If we have to yield, it would be better to yield the whole `nameko.config` object.

Reply:
Here is an example where a test uses `rabbit_config`, but also uses it to write to it an unrelated `serialiser` setting:
https://github.com/nameko/nameko-amqp-retry/blob/master/test/test_retry.py#L458-L475

I'ts one of the places where returning only the passed part of the config may not work.